data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessorConfigTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class AggregateProcessorConfigTest {
+    @Test
+    public void testDefault() {
+        final AggregateProcessorConfig aggregateConfig = new AggregateProcessorConfig();
+
+        assertThat(AggregateProcessorConfig.DEFAULT_WINDOW_DURATION, equalTo(aggregateConfig.getWindowDuration()));
+        assertThat(AggregateProcessorConfig.DEFAULT_DB_PATH, equalTo(aggregateConfig.getDbPath()));
+        assertThat(aggregateConfig.isDbPathValid(), equalTo(true));
+    }
+
+    @Nested
+    class Validation {
+
+        @TempDir
+        File temporaryDirectory;
+        private File file;
+
+        @BeforeEach
+        void setUp() throws IOException {
+            file = new File(temporaryDirectory, UUID.randomUUID().toString());
+            file.createNewFile();
+        }
+
+        @Test
+        void isDbPathValid_should_return_true_for_existing_file() throws NoSuchFieldException, IllegalAccessException {
+            final AggregateProcessorConfig aggregateConfig = new AggregateProcessorConfig();
+            reflectivelySetField(aggregateConfig, "dbPath", file.getAbsolutePath());
```

Comment:
QUES: do we have utilities in data-prepper-api to construct the `AggregateProcessorConfig` with custom field values? Or alternatively, can we use jackson to serialize a map and then read it as AggregateProcessorConfig instance? Asking b/c I am not sure if `reflectivelySetField` will be a standard pattern for other plugin config classes, which seems more like a workaround.

Reply:
Yes the other option is to do this. I decided to use `reflectivelySetField` since it was easier in this case and I only needed to set a single field. This idea stemmed from a PR earlier today where @dlvenable used this method (https://github.com/opensearch-project/data-prepper/pull/830#discussion_r779718439)

Reply:
@graytaylor0 Thanks for the sharing the context.